File: Data Structures & Algorithms/buy-and-sell-crypto-with-cooldown/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length, dp[][] = new int[n + 2][2];
        for(int i = n - 1; i >= 0; --i){
            for(int buy = 1; buy >= 0; --buy){
                if(buy == 1){
                    dp[i][buy] = Math.max(dp[i+1][buy], -prices[i] + dp[i+1][1-buy]);
                }
                else{
                    dp[i][buy] = Math.max(dp[i+1][buy], prices[i] + dp[i+2][1-buy]);
                }
            }
        }

        return dp[0][1];
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res, minCost = 0, 1e9
        for price in prices:
            if minCost > price:
                minCost = price
                continue
            res = max(res, price - minCost)
        return res